### PR TITLE
Added Pktgen test

### PR DESCRIPTION
**Pull request by sandeepkarthikeyan**
Pktgen is a high performance testing tool included in the linux kernel.

It is used to test the TX process of device drivers and NIC.

Signed-off-by: Sandeep K <sandeep@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by sandeepkarthikeyan** [comment]
[job.log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/543761/job.log.txt)

File: io/net/pktgen.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Sandeep K <sandeep@linux.vnet.ibm.com>
+#
+# Based on code by Lucas Meneghel Rodrigues
+# https://github.com/autotest/autotest-client-tests/tree/master/pktgen
```
Comment (PraveenPenguin) [inline io/net/pktgen.py]: i think Martin Bligh is original author  can please rectify this 
```
 Based on code by Martin Bligh <mbligh@google.com>"
                               Copyright: 2008 Google
```

File: io/net/pktgen.py.data/README.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+This program generates network traffic packets. The number of packets, host and other inputs can be given in the yaml file.
```
Comment (PraveenPenguin) [inline io/net/pktgen.py.data/README.txt]: please wrap this in 80 Char

File: io/net/pktgen.py.data/README.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
```
Comment (PraveenPenguin) [inline io/net/pktgen.py.data/README.txt]: remove this white space

File: io/net/pktgen.py.data/README.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+REQUIRED INPUTS:
+ 
+The following inputs are required to run the program else the test will take the default values.
```
Comment (PraveenPenguin) [inline io/net/pktgen.py.data/README.txt]: here also please 80 char

File: io/net/pktgen.py.data/README.txt [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+ 
+The following inputs are required to run the program else the test will take the default values.  
+
+1. Network interface
+2. No of packets to be generated
+3. Clone_skb count
+4. Host physical address
+5. Host IP
+6. Directory to store the results.
+
```
Comment (PraveenPenguin) [inline io/net/pktgen.py.data/README.txt]: please remove all these white spaces

File: io/net/pktgen.py.data/pktgen.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Parameters:
+    eth: "net0"
+    count: "50000"
+    clone_skb: "1"
+    dst_mac: "22:82:8e:e6:94:02"
+    dst_ip: "9.40.192.213"
+    resultsdir: "/root/"
```
Comment (PraveenPenguin) [inline io/net/pktgen.py.data/pktgen.yaml]: can we use some other dir as non root user /root dir not accessible

File: io/net/pktgen.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    DPDK fast packet processing framework
+    '''
+
+    def setUp(self):
+        self.eth = self.params.get("eth", default="eth0")
+        self.count = self.params.get("count", default="50000")
+        self.clone_skb = self.params.get("clone_skb", default="1")
+        self.dst_ip = self.params.get("dst_ip", default="")
+        self.dst_mac = self.params.get("dst_mac", default="")
+        self.resultsdir = self.params.get("resultsdir", default="/root/")
```
Comment (PraveenPenguin) [inline io/net/pktgen.py]: please use some other dir as /root dir not able accessible via non - root user

File: io/net/pktgen.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+        if not os.path.exists('/proc/net/pktgen'):
+            process.system("modprobe pktgen", shell=True)
```
Comment (PraveenPenguin) [inline io/net/pktgen.py]: can you please use ignore_status=True parameter in process.sytem(). if command fail script will error out

File: io/net/pktgen.py [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+        # validating the dst_ip and network interface
+        self.validate_net_interface()
+        self.ping_test()
+
+        # Adding the devices
+        self.log.info("Adding devices")
+        self.pgdev = '/proc/net/pktgen/kpktgend_0'
+        self.pgset('rem_device_all')
+        self.pgset('add_device ' + self.eth)
```
Comment (PraveenPenguin) [inline io/net/pktgen.py]: string concatenation not looks good using `+ ` code can be 

`self.pgset('add_device %s' % self.eth)`

File: io/net/pktgen.py [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+        self.pgset('max_before_softirq 10000')
+
+        # configure the individual devices
+        self.log.info("Configuring the individual devices")
+        self.ethdev = '/proc/net/pktgen/' + self.eth
```
Comment (PraveenPenguin) [inline io/net/pktgen.py]: here also please

File: io/net/pktgen.py [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+        self.pgdev = self.ethdev
```
Comment (PraveenPenguin) [inline io/net/pktgen.py]: can you please code like this 
self.pgdev ='/proc/net/pktgen/ %s' %s self.eth as eliminate extra variable ?

File: io/net/pktgen.py [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+        self.pgset('max_pkt_size 60')
+        self.pgset('dst ' + self.dst_ip)
+        self.pgset('dst_mac ' + self.dst_mac)
+        self.pgset('count %s' % (self.count))
+
+    def test_pktgen(self):
+        self.pgdev = '/proc/net/pktgen/pgctrl'
+        self.pgset('start')
+        output = os.path.join(self.resultsdir, self.eth)
+        process.system("cp %s %s" % (self.ethdev, output), shell=True)
```
Comment (PraveenPenguin) [inline io/net/pktgen.py]: can you please use can you please use  shutil utility for copy file as cp command not looks good ?

File: io/net/pktgen.py [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+
+    def pgset(self, command):
+        file_name = open(self.pgdev, 'w')
+        file_name.write(command + '\n')
+        file_name.close()
+
+        if not self.match_string('Result: OK', self.pgdev):
+            if not self.match_string('Result: NA', self.pgdev):
+                process.system("cat " + self.pgdev, shell=True)
+
```
Comment (PraveenPenguin) [inline io/net/pktgen.py]: here also please ignore_status=True as well remove `+` for string concatenation
code can be 
process.system("cat %s" %self.pgdev, shell=True,ignore_status=True )

**Comment by PraveenPenguin** [comment]
closing this PR as new version of this Test available in PR #126

**Comment by sandeepkarthikeyan** [comment]
[Uploading job.log.txt…]()
Raised new pull request #127

**Pull request closed without merging**